### ISSUE TITLE
Docs: Add more package executors

### ISSUE THREAD

**Issue by Jay-Karia** [eslint/eslint#18777]
### Docs page(s)

One of the doc page:
https://eslint.org/docs/latest/use/command-line-interface#run-the-cli

### What documentation issue do you want to solve?

The commands / scripts uses `npx`.
What about the devs using `pnpm` or `yarn`.

### What do you think is the correct solution?

Add more script blocks for equivalent `pnpm` and `yarn` executors.

### Before:
```shell
npx eslint lib/**
```

### After:
```shell
npx eslint lib/**
```
or
```shell
pnpm dlx eslint lib/**
```
or
```shell
yarn dlx eslint lib/**
```

### Participation

- [X] I am willing to submit a pull request for this change.

### Additional comments

The  equivalents would be:

pnpm - `pnpm dlx ...`
yarn - `yarn dlx ...`

**Comment by amareshsm** [comment]
We can add `pnpm` and `yarn` commands also. Adding multiple options for the same command can clutter the documentation, making it more overwhelming for users who are only interested in one package manager.

IMO Instead of adding every command for `pnpm` and `yarn`, you could include a note or a small section explaining how to run the commands with different package managers. For example:

> Note: If you use `pnpm` or `yarn`, you can replace `npx` with `pnpm dlx` or `yarn dlx`, respectively.

**Comment by harish-sethuraman** [comment]
I feel that adding multiple commands separately just clutters the whole space. I'd be okay with the change if we could provide a tab/dropdown similar to other sites

shadcn UI:
<img width="704" alt="Screenshot 2024-08-14 at 11 38 36 PM" src="https://github.com/user-attachments/assets/80a32dd0-794a-4170-a2c3-27a5f06f9056">
bun: 
<img width="721" alt="Screenshot 2024-08-14 at 11 40 13 PM" src="https://github.com/user-attachments/assets/c2c3710f-1231-46d1-9f0e-2c75aa851e7c">

**Comment by Jay-Karia** [comment]
> We can add `pnpm` and `yarn` commands also. Adding multiple options for the same command can clutter the documentation, making it more overwhelming for users who are only interested in one package manager.
> 
> IMO Instead of adding every command for `pnpm` and `yarn`, you could include a note or a small section explaining how to run the commands with different package managers. For example:
> 
> > Note: If you use `pnpm` or `yarn`, you can replace `npx` with `pnpm dlx` or `yarn dlx`, respectively.

This would be easy for us to implement, and I could not think on how can we implement tabs.

If anyone knows about tab, I am ready to add it.
Until we can go for "note".

Where should we add "note"?

**Comment by aladdin-add** [comment]
the note has already existed:
https://github.com/eslint/eslint/blob/bf96855d7c181648cb0a0e8faf77d707ddd4725f/docs/src/use/command-line-interface.md?plain=1#L48

A pr to improve it is welcome!

**Comment by Jay-Karia** [comment]
Here is an improved note I can think of:

**Note:** You can also use alternative package managers such as [Yarn](https://yarnpkg.com/) or [pnpm](https://pnpm.io/) to run ESLint. For `pnpm` add prefix `pnpm dlx` and for `yarn` and prefix `yarn dlx`.

**Comment by aladdin-add** [comment]
Generally, we are reluctant to include usage details of other projects in our documentation due to the maintenance and synchronization costs. However, since pnpm/yarn has become quite stable with little likelihood of changes in usage, I think it might be feasible in this case. thoughts?  @eslint/eslint-team 

@Jay-Karia please feel free to send a PR, we can discuss the details in there. :)

**Comment by nzakas** [comment]
> I feel that adding multiple commands separately just clutters the whole space. I'd be okay with the change if we could provide a tab/dropdown similar to other sites

@harish-sethuraman I agree. This is a much better approach. We do already have a tabs component in our library (https://eslint.org/docs/latest/component-library/code-tabs). We'd just need to create a shortcode to embed into the docs.
